src/main/java/seedu/budgetbuddy/commons/ExpenseList.java

```diff
@@ -154,42 +152,72 @@ public double calculateTotalExpenses() {
         return totalExpenses;
     }
 
+    private boolean checkBudgetBeforeAddingExpense(String category, double amountAsDouble) {
```

Comment:
The method name checkBudgetBeforeAddingExpense implies a check but doesn't clearly indicate what true or false return values signify. Would renaming the method to something like isExpenseOverBudget be better? This would help to directly reflect that true means the expense would exceed the budget.